certification/pyxis/submit.go

```diff
@@ -68,6 +71,16 @@ func (p *pyxisClient) SubmitResults(ctx context.Context, certInput *Certificatio
 		if err != nil {
 			return nil, fmt.Errorf("could not get image: %v", err)
 		}
+
+		// checking to see if the original value is certified and the previous value is not certified,
+		// this would indicate that a partner is running preflight again, and during the first run there was a timeout/error
+		// in a check that interacts with pyxis and we need to correct the certified value for the image
+		if certified && !certImage.Certified {
+			certImage, err = p.updateImage(ctx, certImage.ID, certImage.Architecture)
```

Comment:
I think it makes more sense to just pass in  &certImage. If more fields are added later, you won't have to change this, just updateImage.

Reply:
Are you saying build the new struct outside of the `updateImage` method and pass that in? I thought about that, but in updateProject we build the struct in the method, so I was trying to be consistent. Both ways make sense to me, and I have no strong opinion either way.

Reply:
I left a comment elsewhere, but I'm of the opinion that `updateImage` is probably not how we want to refer to this method given its context, where by it only updates certified to true.

I'm not sure if passing in a certImage makes sense to me given that context. Do we not already send an Update call to Pyxis in this workflow? Or is the expectation that we're always sending a Create call? If we're looking to build a separate workflow for Update, then maybe we need to detect earlier in Submit whether we need to go down an Update (because the image already exists) or Create (because the image doesn't already exist) workflow and handle things independently?

Just a few thoughts. Not sure where I land on it, but I'm less close to the issue at hand.

Reply:
Since we don't know if a user is running preflight for the first time or nth time for a given digest/arch combination, we built the submit flow for `images` and `rpm` manifest, to first call `create`, then if we get a `conflict` then we would call a `get`.

This is just an extension of that logic, with the addition that we need to correct the certification value from a previous run. Do we buy anything from switching to a `get` first? then falling back to create if `not found`? That seems like a larger refactor and a higher chance of getting something wrong, then we also should switch RPM manifest calls to follow that logic so we are consistent.

That's just my take, I could be being short sighted.